src/components/Main/Main.js

```diff
@@ -229,7 +233,8 @@ export class MainComponent {
     this.player = new MediaPlayerComponent(this.playerPoint, {
       onDialogOpen: this.handleDialogOpen.bind(this),
       onPlayerChangeState: this.handlePlayerChangeState.bind(this),
-      onLegalOptionClick: this.handleSetInfo.bind(this)
+      onLegalOptionClick: this.handleSetInfo.bind(this),
+      onAddSong: this.handleAddSong.bind(this)
```

Comment:
there is no need for this handleAddSong. You can just directly this.table.addSong.bind(this.table)